README.md

```diff
@@ -124,9 +124,11 @@ class CountingDelegate: HTTPResponseDelegate {
         // this is executed when we receive HTTP Reponse head part of the request (it contains response code and headers), called once
     }
 
-    func didReceivePart(_ buffer: ByteBuffer) {
+    func didReceivePart(_ buffer: ByteBuffer) -> EventLoopFuture<Void>? {
```

Comment:
Returning an optional future seems unusual to me. Could we achieve the same thing by passing in an optional promise?

Reply:
See main discussion, having promise here could be error-prone